composer/datasets/utils.py

```diff
@@ -140,20 +143,35 @@ def add_vision_dataset_transform(dataset: VisionDataset, transform: Callable, is
 
     transform_added_logstring = textwrap.dedent(f"""\
         Transform {transform} added to dataset.
-        Dataset now has the following transforms: {dataset.transform}""")
+        Dataset now has the following transforms: {dataset.transforms}""")
 
-    if dataset.transform is None:
+    if is_target_transformed:
+        if dataset.transforms is None:
+            dataset.transforms = transform
+            log.warning(transform_added_logstring)
+
+    elif dataset.transform is None:
         dataset.transform = transform
         log.warning(transform_added_logstring)
-    elif isinstance(dataset.transform, transforms.Compose):
-        insertion_index = len(dataset.transform.transforms)
-        for i, t in enumerate(dataset.transform.transforms):
+
+    data_transforms = dataset.transforms if is_target_transformed else dataset.transform
+
+    if isinstance(data_transforms, transforms.Compose):
+
+        insertion_index = len(data_transforms.transforms)
+        for i, t in enumerate(data_transforms.transforms):
             if isinstance(t, transforms.ToTensor):
                 insertion_index = i
                 break
         if is_tensor_transform:
             insertion_index += 1
-        dataset.transform.transforms.insert(insertion_index, transform)
+
+        if is_target_transformed:
+            assert dataset.transforms is not None
+            dataset.transforms.transforms.insert(insertion_index, transform)
```

Comment:
The difference between the two cases here is hard to spot. Maybe a comment could point it out.

Reply:
Deleted this feature, so marking as resolved